src/win32.c

```diff
@@ -1032,4 +1032,31 @@ gchar *win32_get_user_config_dir(void)
 	return g_build_filename(g_get_user_config_dir(), "geany", NULL);
 }
 
+
+void win32_make_argc_and_argv_in_utf8(gint *pargc, gchar ***pargv)
+{
+	int num_arg;
+	LPWSTR *szarglist = CommandLineToArgvW(GetCommandLineW(), &num_arg);
+	char **utf8argv = g_new0(char *, num_arg + 1);
+	int i = num_arg;
+	while(i)
+	{
+		i--;
+		utf8argv[i] = g_utf16_to_utf8((gunichar2 *)szarglist[i], -1, NULL, NULL, NULL);
```

Comment:
@codebrainz myeah, theoretically it might, but in practice it won't change (nobody can afford that honestly), and that Windows API is a lot more annoying to use for us.  But if it does handle the invalid Unicode filenames and `g_utf16_to_utf8()` doesn't, it might be worth it.